package/src/utilities/add-devtoolbar-framework-app.ts

```diff
@@ -35,9 +17,53 @@ export type AddDevToolbarFrameworkAppParams = {
 	style?: string;
 } & Pick<
 	HookParameters<"astro:config:setup">,
-	"addDevToolbarApp" | "updateConfig" | "injectScript"
+	"addDevToolbarApp" | "updateConfig" | "injectScript" | "config"
 >;
 
+const frameworkDependencies: Record<SupportedFrameworks, string[]> = {
+	preact: ['preact'],
+	react: ['react', '@vitejs/plugin-react'],
+	svelte: ['svelte'],
+	solid: ['solid-js'],
+	vue: ['vue'],
+}
+
+// Returns the path of the AIK folder
+function getNearestPackageJson(directory: string): string {
+	const { resolve } = createResolver(directory);
+	const packageJsonPath = resolve('./package.json');
+
+	if (existsSync(packageJsonPath)) {
+		return resolve();
+	}
+
+	const parentDir = dirname(directory);
+	if (parentDir === directory) {
+		// Reached the root directory without finding a package.json
+		throw new AstroError("Can't find package.json. This should never, EVER get hit. If you ever get this error, sorry we goofed.");
+	}
+
+	return getNearestPackageJson(parentDir);
+}
+
+function getMissingFrameworkDependencies(framework: SupportedFrameworks, config: HookParameters<"astro:config:setup">["config"]): string[] {
+	const projectRoot = config.root.pathname;
+	const aikRoot = getNearestPackageJson(import.meta.url);
```

Comment:
This should not be AIK's `import.meta.url`, it should be the `import.meta.url` of the integration that is using AIK.

That is the tricky bit that maybe wasn't clear from Discord.

We can either receive it as a parameter (may be optional and in that case require the dependency on the root) or we can try to infer it dynamically using the stack frames

Reply:
Or we can always require the dependency to be present on the project for now and relax this requirement later

Reply:
I think that would make things easier. As long as it's documented, it's fine!